### DISCUSSION TITLE
Configure the number to rem values.

### DISCUSSION THREAD

**Discussion by christhofer** [unocss/unocss#3082] [Ideas]
Currently using preset Uno/Wind, we have the spacing like
`m-4` gives 1rem / 16px
`m-6` gives 1.5rem / 24px

Can we make the multiplier to be configurable? I want to custom the number, like
`m-4` gives 0.25rem / 4px
`m-6` gives 0.375rem / 6px
`m-16` gives 1rem / 16px 
`m-30` gives 1.875rem / 30px
Basically the number equals to the pixel, but the CSS uses rem.

I looked around the source code, the calculation is hardcoded [here](https://github.com/unocss/unocss/blob/44b7ebb91c9460f3d8d23f330213b7d0aac7b991/packages/preset-mini/src/_utils/handlers/handlers.ts#L56C66-L56C66). 
Can we have the `4` here to be configurable, allows us to change it to `16`.
```js
return unit ? `${round(num)}${unit}` : `${round(num / 4)}rem`
```

I know we can do `m-16px` and `m-30px`, but I want to use rem, and calculating `30/16` to get `m-1.875rem` is not practical.

I also know we can use rules:
```js
rules: [
  [/^m-(\d+)$/, ([, d]) => ({ margin: `${Number(d) / 16}rem` })],
]
```
But then we also need to change the rules for `mt-`, `mb-`, `mr-`, `mt-`, `w-`, `min-w-`, and so on. It's so many. And the `space-x-` `space-y-` is not that simple.

I'm willing to make the pull request if my request here got a green light.

My idea is to add a property to config `theme`, maybe like `remDivider` _(any idea is appreciated)_
```ts
theme: {
  remDivider: 16,
}
```

**Comment by zyyv** [comment]
You can convert generated css in `postprocess` hook as a workaround.
Refer to [Playground](https://unocss.dev/play/?html=DwEwlgbgBA7gTgQwA5IKZwHwCgpVJKAWwFoAWI4gNgoEZqSAmABigwFUA7AewGEBlPsAD04CNmGjsQA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAJIBjAlT6UaMAKpYIcAL5x0UCCDgByNnOFUqGgFAHUAD0iwUGAIZsANvDSYc%2BQsCIAKBAbhwqAC2gYYQ4qAC5Ebx84AHcoKzAwVChwjWiAWgBGAAYsgDdouD9MnNy-ZVtTOGAYVBAqNOFULBqoOAArNipOHgamlo0SSIVBn34pMLgAbUjRyVpZCHcAShGosdoGJlYOLm5l1dmBGBExTxmoqmErCvCMgDoAJgOo4WQsFL9GMDCAeh-qEB3fw-AbnBQrSIAXVWkC6FAgjT04WmUTg7h2tiWcAAvAA%2BCKonwYu59ZjUO7oaAAUSswj87ncwCxeIJhJ8oiwXTguWubFQOKqkwykPOqK4aJg3ESEHQ3N5-Oxis0XWYWCIGjgADJNXAfu40gB%2BSYAHTuxuQkIA1EtKCAfkQ7jUuu4ebY%2BUslqLCcAhZCBa6%2BXdKGBbLTUO49YaTWaLdbbfayO4APp8DLM-EAAwAJAgwBldXAACwKW0Zz1s8Fg1bQgyVoA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

**Reply by christhofer** [reply]
That seems very hacky.

It also converts `w-2rem` to 0.5rem.
`text-2xl` converted to 0.375rem

**Reply by christhofer** [reply]
Postprocess also doesn't seem to work on `@apply`
```css
.custom {
  @apply p-16;
}
```

still outputs
```css
.custom{padding:4rem;}
```
[playground](https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIDuAnBAHPAU23QD4AoKKUSWRFDAWwFoAWKPN7QxqAF0IAPPswDMAJn4B7TqzyCyAVQB2UgMIBlDcAD04CBV36KQA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAJIBjAlT6UaMAKpYIcAL5x0UCCDgByNnOFUqGgFChIsOEwCGWKumghUUACLBKwzgDdqy1eo0ABbRC6VAD0FlY2UHZQALTIzqiuwB76BgaoAB4m8Gjo5mwANtkY2HgExAAUCAZwcFQAFtAwwhxUAFyI1TVwAO5Q5mBg9u0a3dEAjAAME27dcHXjU251yvkZcMAwqCBU0cKoWJtQcABWbFScPLv7hxoknQp3NfxSbXAA2p1PkrSyEOUAlI84ABdIFhay2eyvD5dMx9cKQxzxRLJAFkYLBOD7cwsVZwXz9MD5bjKaBwIKdUGdSDnCiBajQz5wcoQFjHf5wAC8AD4OrCalxmcFom8ADrIYEAalFAHUAPwAEmCRAAdJtziy2SqaKtXNAOZzDcpzPkaByqvyupQYGwoFg4KzjkyagpnQ6tddmNQVREAKLmYR1crlYAG3kWy3ksTwNwmtioLnrN5jYFugXoZkwbiDCAZ2P5eNco0ac7MLBEDRwABkVbgwXK0TlYpV4ql-0oIGVauoMHK%2Bfj-3Naa6wGTwMT-dQKsoRIDqHK9cbzdbkvbW2VZHKAH0%2BGMw3AAAYKhBgMZ1uAAFgUHYP-2HrstCjvsIelLuT4MQA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YEMABQYtJlxgA3tLBgAArWEAbAJ5hBAWgCMANgDc0gL5A&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

**Reply by zyyv** [reply]
Postprocess is not effective for `transformers`, it only processes the extracted and output unitility.

**Comment by christhofer** [comment]
I can't find a  way to read the config from inside the `rem` function, without passing it as a 2nd argument...
I guess it's not possible...?

**Reply by christhofer** [reply]
Ended up I getting the result I wanted by adding more checks to Zyvv's workaround to prevent affecting 2rem or 2xl classes.

```js
  postprocess: [
    (obj) => {
      if (/-[\d]+\W?$/g.test(obj.selector) === false) {
        return obj
      }
      obj.entries.forEach((i) => {
        const value = i[1]
        if (typeof value === 'string' && /(-?[\.\d]+)rem/g.test(value)) {
          i[1] = value.replace(/(-?[\.\d]+)rem/g, (_, p1) => `${p1 / 4}rem`)
        }
      })
    },
  ],
```
[Playground](https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIDuAnBAHPAU23QD4AoKKUSWRFDAWwFoAWKPN7QxqAF0IAPPswDMAJn4B7TqzyCyAVQB2UgMIBlDcAD04CBV36KQA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmANHGFKgM6owCCMMUwARgK4zDoCeZF1tAJIBjAlT6UaMAKpYIcAL5x0UCCDgByNnOFUqGgFAHUAD0iwUGAIZsANvDSYc%2BQsCIAKBAbhwqAC2gYYQ4qAC5Ebx84AHcoKzAwVChwjWiAWgBGAAYsgDdouD9MnNy-ZVtTOGAYVBAqNOFULBqoOAArNipOHgamlo0SSIVBn34pMLgAbUjRyVpZCHcAShGosaFRLCpPGaiqYSsK8IyAOgAmVai4YWQsFL9GMDCAemfqEBP-Z4HdhRXIgC6q0gXQoEEaenC0yu7ggLDaSzgAF4AHwRK4%2BLhwdzPNKTAA6yABAGp8QB1AD8ABJnkQTjUurD4Z9UBVhDBoIikdzlIcaIivBiopQYGwoFg4HC2rsfAoZZLmX1mNQTuhoABRKzCPzudzALlowVC65ieC5Q5sVDIqqTDIA%2BWY9DYmDcRIQJ3m2yW5E8jRdZhYIgaOAAMhDcGe7jSFIJJ0JJKWlBAtPp1Bg7k9lqWAodUWAtoB1szqBOlDAti1qBxUZj%2BLjROJidqtLI7gA%2BnwMga4AADKkIMAZCNwAAsCiTPaWublQr%2Bv1WQIMc6AA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwF8g)

Haven't tested it on a big project. I think this is a bad idea and error-prone.

**Comment by asd8855** [comment]
+1
